src/lib/toolbar.ts

```diff
@@ -324,8 +324,11 @@ export const toolbarConfig: Record<ToolbarModeName, ToolbarMode> = {
           icon: 'arc',
           status: 'available',
           disabled: (state) =>
-            !isEditingExistingSketch(state.context) &&
-            !state.matches({ Sketch: 'Tangential arc to' }),
+            (!isEditingExistingSketch(state.context) &&
+              !state.matches({ Sketch: 'Tangential arc to' })) ||
+            state.matches({
+              Sketch: { 'Rectangle tool': 'Awaiting second corner' },
+            }),
```

Comment:
Interesting, I would have figured this wouldn't be necessary because of the first half of this condition?

Reply:
Oh right let me remove this, I think this is bleed over from the circle PR anyway, plus I think your right, doesn't seem right let me check.

Reply:
Yeah change it back.